src/filesystem/impls/filer/FilerFileSystem.js

```diff
@@ -347,35 +347,34 @@ define(function (require, exports, module) {
         });
     }
 
-    function unlink(path, callback) {
-        path = decodePath(path);
-
-        fs.stat(path, function(err, stats) {
+    function _rmfr(path, callback) {
+        // Regardless of whether we're passed a file or dir path, recursively delete it all.
+        fs.rm(path, {recursive: true}, function(err) {
             if (err) {
                 callback(_mapError(err));
                 return;
             }
 
-            // Deal with dir vs. file
-            var fnName = stats.type === "DIRECTORY" ? 'rmdir' : 'unlink';
-            fs[fnName](path, function(err) {
-                // TODO: deal with the symlink case (i.e., only remove cache
-                // item if file is really going away).
-                BlobUtils.remove(path);
+            // TODO: deal with the symlink case (i.e., only remove cache
+            // item if file is really going away).
+            BlobUtils.remove(path);
```

Comment:
It doesn't purge the cache first, but I can make it do that.  The only thing I'm not sure about is whether or not this will break anything, since there might be URLs in use that will suddenly go away.  It might make more sense to just leave them in the cache, since they will be destroyed when the app is reloaded, and won't cause any problem.